waspc/src/Wasp/Generator/Common.hs

```diff
@@ -12,21 +11,30 @@ import qualified Wasp.SemanticVersion as SV
 -- | Directory where the whole web app project (client, server, ...) is generated.
 data ProjectRootDir
 
--- | Node version that node packages generated by this generator expect.
-nodeVersion :: SV.Version
-nodeVersion = SV.Version 16 0 0 -- Latest LTS version.
-
-nodeVersionBounds :: SV.VersionBounds
-nodeVersionBounds = SV.BackwardsCompatibleWith nodeVersion
-
-npmVersion :: SV.Version
-npmVersion = SV.Version 8 0 0 -- Latest LTS version.
-
-npmVersionBounds :: SV.VersionBounds
-npmVersionBounds = SV.BackwardsCompatibleWith npmVersion
+-- | Range of node versions that node packages generated by this generator work correctly with.
+nodeVersionRange :: SV.Range
+nodeVersionRange =
+  SV.rangeFromVersionsIntersection
+    [ (SV.BackwardsCompatibleWith, latestLTSVersion),
+      (SV.LessThanOrEqual, latestLTSExactVersionThatWeKnowWorks)
+    ]
+  where
+    latestLTSVersion = SV.Version 16 0 0
+    -- There is a bug in node 16.15.1 (more correctly, in npm 8.11.0 that comes with it)
+    -- that messes up how Wasp uses Prisma. That is why we limited ourselves to <=16.15.0 for now.
+    -- Bug issue on NPM CLI repo: https://github.com/npm/cli/issues/5018 .
+    latestLTSExactVersionThatWeKnowWorks = SV.Version 16 15 0
+
+-- | Range of npm versions that Wasp and generated projects work correctly with.
+npmVersionRange :: SV.Range
+npmVersionRange =
+  SV.rangeFromVersionsIntersection
+    [ (SV.BackwardsCompatibleWith, latestLTSVersion),
+      (SV.LessThanOrEqual, latestLTSExactVersionThatWeKnowWorks)
+    ]
+  where
+    latestLTSVersion = SV.Version 8 0 0 -- Goes with node 16
+    latestLTSExactVersionThatWeKnowWorks = SV.Version 8 5 5 -- Goes with node 16.15.0
 
 prismaVersion :: SV.Version
```

Comment:
then this should be `supportedPrismaVersion` hmm. Maybe `Common.hs` is not the best (for sure it isn't, it shouldn't exist according to uncle bob :D) place for this, we could extract version stuff under e.g. `SupportedVersions.hs`?